src/lib/autocomplete/autocomplete.component.ts

```diff
@@ -0,0 +1,175 @@
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import {
+    AfterContentInit,
+    AfterViewInit,
+    Component,
+    ContentChild,
+    forwardRef,
+    Input,
+    OnDestroy,
+    OnInit,
+    ViewChild,
+} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+import { Observable, Subject } from 'rxjs';
+import { debounceTime, delay, mergeMap, takeUntil, tap } from 'rxjs/operators';
+
+import { RockInputDirective } from '../core/directives/input.directive';
+import { RockOverlayDirective } from '../core/directives/overlay.directive';
+import { RockErrorComponent } from '../error/error.component';
+import { RockAutocompleteList } from './autocomplete-list';
+import { RockAutocompleteValue } from './autocomplete-value';
+
+let uniqueId = 0;
+
+@Component({
+    selector: 'rock-autocomplete',
+    templateUrl: './autocomplete.component.html',
+    styleUrls: [ './autocomplete.component.scss' ],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            multi: true,
+            useExisting: forwardRef(() => RockAutocompleteComponent),
+        },
+    ],
+})
+export class RockAutocompleteComponent implements OnInit, AfterContentInit, AfterViewInit, ControlValueAccessor, OnDestroy {
+
+    public isOpen = false;
+    public hasError = false;
+    public isDisabled = false;
+    public isRequired = false;
+    public displayValue: string;
+    public options: Array<RockAutocompleteList> = [];
+
+    private inputListener: Subject<string> = new Subject<string>();
+    private destroyed: Subject<boolean> = new Subject<boolean>();
+
+    @Input()
+    public id = `rock-autocomplete-${ ++uniqueId }`;
+
+    @Input()
+    public label: string;
+
+    @Input()
+    public name: string;
+
+    @Input()
+    set required(required: boolean) {
+        this.isRequired = coerceBooleanProperty(required);
+    }
+    get required(): boolean {
+        return true;
```

Comment:
If it is possible to `set required` should the `get` not return `this.isRequired`?